src/ragas/testset/persona.py

```diff
@@ -0,0 +1,150 @@
+from ragas.testset.graph import KnowledgeGraph, Node
+
+logger = logging.getLogger(__name__)
+
+
+def default_filter(node: Node) -> bool:
+
+    if (
+        node.type.name == "DOCUMENT"
+        and node.properties.get("summary_embedding") is not None
+    ):
+        return True
+    elif (
+        node.type.name == "CHUNK"
+        and node.properties.get("topic_description_embedding") is not None
+    ):
+        return random.random() < 0.1
+    else:
+        return False
+
+
+class SummaryInput(BaseModel):
+    summaries: t.List[str]
+
+
+class Persona(BaseModel):
+    name: str
+    role_description: str
+
+
+class PersonasList(BaseModel):
+    personas: t.List[Persona]
+
+    def __getitem__(self, key: str) -> Persona:
+        for persona in self.personas:
+            if persona.name == key:
+                return persona
+        raise KeyError(f"No persona found with name '{key}'")
+
+
+# Define the prompt class
+class PersonaGenerationPrompt(PydanticPrompt[SummaryInput, PersonasList]):
+    instruction: str = (
+        "Using the provided summaries, generate one persona for each summary who might "
+        "interact with the content. For each persona, include a unique name "
+        "and a brief role description of who they are."
+    )
+    input_model: t.Type[SummaryInput] = SummaryInput
+    output_model: t.Type[PersonasList] = PersonasList
+    examples: t.List[t.Tuple[SummaryInput, PersonasList]] = [
+        (
+            SummaryInput(
+                summaries=[
+                    "Guide to Digital Marketing explains strategies for engaging audiences across various online platforms.",
+                    "Data Privacy Essentials discusses principles for safeguarding user data and complying with privacy regulations.",
+                    "Introduction to Project Management covers key methods for planning, executing, and monitoring projects.",
+                ]
+            ),
+            PersonasList(
+                personas=[
+                    Persona(
+                        name="Digital Marketing Specialist",
+                        role_description="Focuses on engaging audiences and growing the brand online.",
+                    ),
+                    Persona(
+                        name="Data Privacy Officer",
+                        role_description="Ensures the organization's compliance with data protection laws.",
+                    ),
+                    Persona(
+                        name="Project Manager",
+                        role_description="Oversees project timelines and ensures tasks are completed efficiently.",
+                    ),
+                ]
+            ),
+        )
+    ]
+
+
+@dataclass
+class PersonaGenerator:
+
+    llm: BaseRagasLLM
+    num_personas: int = 5
+    prompt: PydanticPrompt = PersonaGenerationPrompt()
+    filter_nodes: t.Callable[[Node], bool] = field(
+        default_factory=lambda: default_filter
+    )
```

Comment:
add this as part of `PersonaList`?

so the interaction will become `PersonList.generate_from_kg(num_persona, kg, llm, prompt, filter)`
?